app-backend/tile/src/main/scala/routes/ToolRoutes.scala

```diff
@@ -81,64 +71,67 @@ object ToolRoutes extends LazyLogging {
     }.toMap
   }
 
-  def root(implicit db: Database): Route =
-    pathPrefix(Segment / "ndvi-diff-tool"){ organizationId =>
-      (pathEndOrSingleSlash & get & rejectEmptyResponse) {
-        complete("model JSON")
-      } ~
-      pathPrefix(IntNumber / IntNumber / IntNumber){ (z, x, y) =>
-        parameter(
-          'part.?,
-          'LC8_0, 'LC8_1,
-          'class0.?("0.1:0;99999999:1.0"), // Make this big so that all values are caught
-          'class1.?("0.1:0;99999999:1.0"),
-          'geotiff.?(false),
-          'cm.?
-        )
-        { (partId, p0, p1, class0, class1, geotiffOutput, colorMap) =>
-          complete {
-            val varMap = Map(
-              'LC8_0 -> { (z: Int, x: Int, y: Int) =>
-                Mosaic.raw(UUID.fromString(p0), z, x, y).value},
-              'LC8_1 -> { (z: Int, x: Int, y: Int) =>
-                Mosaic.raw(UUID.fromString(p1), z, x, y).value})
-
-            val model: Op = partId match {
-              case Some("ndvi0") => ndvi0
-              case Some("ndvi1") => ndvi1
-              case Some("class0") => reclass0(parseBreakMap(class0))
-              case Some("class1") => reclass1(parseBreakMap(class1))
-              case _ => ndviDiff(parseBreakMap(class0), parseBreakMap(class1))
-            }
-
-            logger.debug(s"model prior to param binding: $model")
-
-            // TODO: Inspect what variables are actually required, only fetch those tiles
-            val futureTiles = varMap.map { case (sym, tms) =>
-              tms(z, x, y).map { maybeTile => sym -> maybeTile }
-            }
-
-            Future.sequence(futureTiles).map { tiles =>
-              // construct a list of parameters to bind to the NDVI-diff model
-              val params = tiles.map { case (name, maybeTile) =>
-                Op.Var(name) -> Op.Unbound(maybeTile)
-              }.toMap
-
-              logger.debug(s"Params to bind to model: $params")
-              val assignedModel = model.bind(params)
-              logger.debug(s"model after param binding: $assignedModel")
-              assert(assignedModel.fullyBound) // Verification that binding completed
+  def root(
+    source: (RFMLRaster, Int, Int, Int) => Future[Option[Tile]]
+  ): Route =
+    pathPrefix(JavaUUID / JavaUUID){ (orgId, toolId) =>
```

Comment:
What is the org id doing here? At this point with authentication we should have the user I think which has the organization id.